src/inmanta/module.py

```diff
@@ -132,6 +132,13 @@ def __contains__(self, version: str) -> bool:
     def __str__(self) -> str:
         return str(self._requirement).replace("-", "_")
 
+    def __hash__(self) -> int:
+        hashCmp = (
+            self._requirement.hashCmp,
```

Comment:
why not `hash(self._requirement)` or `self._requirement.__hash__() `

Reply:
Because the hash value has to be the same for two objects when the `__eq__` method returns true. The implementation of `__eq__` in this class deviates from the implementation of the `__eq__` method in the parent (`Requirement`) by the fact that the `project_name` should also be equal. That's why the hash takes `project_name` into account. To be honest I am not sure why this is important. Maybe @sanderr can explain?

Reply:
I haven't looked at the context here yet but generally the hash needs to be equal for equal objects because it is used as a key in `Set` and `Dict` (and possibly other data structures as well). If `one == other => hash(one) == hash(other)` doesn't hold, it's impossible to use the hash to find an instance in such a data structure. Suppose you try `x in myset`, to efficiently determine this, `hash(x)` is used as a key to a bucket of objects, within which the slower `__eq__` is used.
(This is from the top of my head, I might have gotten some details wrong but the general reasoning should hold.)

Reply:
The parent doesn't use the `project_name` property because uses `key`, which is equal to `project_name.lower()`. In other, the parent does a case insensitive match on `project_name`, while the subclass does a case sensitive match.

Reply:
I think we should enforce that `InmantaModuleRequirement` doens't accept any requirements with a capital letter in it. Any opinions.

Reply:
I have no strong opinion on whether we should allow capital letters in module names. As for using `key` instead of `project_name`, I think that would be a good idea in any case (we're just talking about the `__eq__` method, right? I'm even wondering why the `project_name` check is there at all, I think it would be safe to just delegate `__eq__` to `self._requirement`.

Reply:
I excluded the check on `project_name` from the `__eq__()` method.